### ISSUE TITLE
Notification must set user model id on win32

### ISSUE THREAD

**Issue by txthinking** [asticode/go-astilectron#146]
I did not find the setappusermodelid api on godoc.

https://electronjs.org/docs/api/app#appsetappusermodelidid-windows

BTW, can we set app id custom like com.mydomain.app

Thanks

**Comment by asticode** [comment]
@txthinking ok for `app.setAppUserModelId(id)`.

But regarding your second question, can you point to the Electron corresponding doc?

**Comment by txthinking** [comment]
Sorry for the unclear description.
https://www.electron.build/configuration/configuration#configuration

**Comment by asticode** [comment]
thanks!

**Comment by CarsonSlovoka** [comment]
I spent some time and found the solution and shared it with the people who needed it.

----

I using the [astilectron 0.27| astilectron: 0.51.0 | Electron 11.4.3](https://github.com/asticode/go-astilectron/blob/5317e8218ad3c/astilectron.go#L16-L17)

I can wait for [provision] to finish downloading and processing all the related files and then modifying [index.js](https://github.com/asticode/astilectron/blob/a8c3f044b1feecab7f18840f6769ed7ec576da1d/index.js#L28-L30)

> where "index.js" is located on:  https://github.com/asticode/go-astilectron/blob/e7796e5/paths.go#L62
>  
> if os:windows default path: `%appdata%/[appName]/vendor/astilectron/index.js`

[provision]: https://github.com/asticode/go-astilectron/blob/e7796e5811c6df568618862bf45de5bce2111fc6/astilectron.go#L170

```js
// App is ready
function onReady () {
    app.setAppUserModelId("com.mydomain.app");  //  👈  Add this line.  default: electron.app.Electron. 
    // Init
    const screen = electron.screen
    Menu.setApplicationMenu(null)
    ...
```

But I think this method is still very bad, I hope that the code can be changed directly in the go🤔

----

see also [how can you overwrite or remove the signature "electron.app.Electron" from the desktop notification](https://stackoverflow.com/q/64088839/9935654)

**Comment by asticode** [comment]
Yeah, I think that an `AppUserModelID` should be added in the [Options](https://github.com/asticode/go-astilectron/blob/master/astilectron.go#L63).

I won't have time to work on this anytime soon, but if someone is looking to contribute, feel free to let me know and I'll guide you to make a PR.

**Comment by CarsonSlovoka** [comment]
I'm trying to use the event to change the ID, 

I with it can add on [notificationCreate](https://github.com/asticode/astilectron/blob/a8c3f04/index.js#L150)

but it **doesn't work**. It seems that the event has to be determined after the end of [onReady](https://github.com/asticode/astilectron/blob/a8c3f044b1feecab7f18840f6769ed7ec576da1d/index.js#L28-L29). Even if the hard coding can't change the ID later, for [example](https://github.com/asticode/astilectron/blob/a8c3f044b1feecab7f18840f6769ed7ec576da1d/index.js#L57-L62)

```js
    rl.on('line', function(line){
        app.setAppUserModelId("hello world")  // 👈 still not work.
        // Parse the JSON
        let json = JSON.parse(line)

        // Switch on event name
        let window;
```

I don't know if my inference is wrong or not, but the result of my experiment is indeed like this.

----

I noticed that [atilectron](https://github.com/asticode/astilectron) is not a big repository, I want to use embed<sup>[go1.16 embed](https://pkg.go.dev/embed)</sup> in [go-astilectron](https://github.com/asticode/go-astilectron) to embed it directly so that there is no need to maintain two repositories, and even in the js directly using [text/template](https://pkg.go.dev/text/template#hdr-Actions), we can set some initialization place to supply the user to write the code directly into the js, will become very flexible.

Would you accept the pull request like it?
